.github/workflows/test-integration-aws_lambda.yml

```diff
@@ -25,7 +25,18 @@ env:
     ${{ github.workspace }}/dist-serverless
 
 jobs:
+  authorize:
+    name: Determine environment
+    environment:
+      ${{ github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository &&
+      'external' || 'internal' }}
+    runs-on: ubuntu-latest
+    steps:
+      - run: true
```

Comment:
this is not clear what this does -- I expected a label being required here and I don't see it

as it is written right now it is unsafe as it checks out the pull request and then executes arbitrary code from it

Reply:
The idea is to use environments instead of labels to control which PRs are allowed to be checked out and executed. The prerequisite is that the repo must be configured to have an `external` environment such that any `environment: external` workflows [require a review](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#required-reviewers) from a Sentry employee before they're allowed to be run.

So technically this should be better than using a label, since it revokes any existing workflow approvals automatically on new code changes without us needing to introduce another step to check for code changes and remove the label, and there is no risk of a race condition between granting the approval/label and new code changes being pushed.

Reply:
More context: https://github.com/iterative/cml/pull/574#issuecomment-1188650440

Reply:
hmm nice if it works -- we should try this out in a separate repo first with less sensitive credentials

if this is the github feature I think it is we still might not want to do it this way though -- iirc it sends an email to all repository collaborators (read: the entire company) without a way to opt out

Reply:
I tried this out on https://github.com/sentrivana/playground: here's the [ci.yaml](https://github.com/sentrivana/playground/blob/main/.github/workflows/ci.yaml), the environment definition:

![Screenshot 2023-11-10 at 13 37 58](https://github.com/getsentry/sentry-python/assets/131587164/1d9ab30e-5e84-4b1c-961d-f7b4de6875bb)

And here's a fork PR against the repo: https://github.com/sentrivana/playground/pull/2

> if this is the github feature I think it is we still might not want to do it this way though -- iirc it sends an email to all repository collaborators (read: the entire company) without a way to opt out

Ugh. Will test this some more to see if that's the case. If it is, will change this to use a label instead.

Thanks for the feedback!